src/client/visualizations/scatterplot/scatterplot.tsx

```diff
@@ -23,11 +23,53 @@ import {
   VisualizationProps
 } from "../../views/cube-view/center-panel/center-panel";
 
+import * as d3 from "d3";
+import { Datum } from "plywood";
+import { ConcreteSeries } from "../../../common/models/series/concrete-series";
+import { selectFirstSplitDatums } from "../../utils/dataset/selectors/selectors";
 import "./scatterplot.scss";
 
-const Scatterplot: React.SFC<ChartProps> = () => {
-  return <div className="scatterplot-container">
-    <h2>Scatterplot will be here</h2>
+import { Stage } from "../../../common/models/stage/stage";
+import { GridLines } from "../../components/grid-lines/grid-lines";
+import { pickTicks } from "../../utils/linear-scale/linear-scale";
+import { Point } from "./point";
+import { XAxis } from "./x-axis";
+import { YAxis } from "./y-axis";
+
+function getExtent(data: Datum[], series: ConcreteSeries): number[] {
+  const selectValues = (d: Datum) => series.selectValue(d);
+  return d3.extent(data, selectValues);
+}
+
+const Scatterplot: React.SFC<ChartProps> = ({ data, essence, stage }) => {
+  const [xSeries, ySeries] = essence.getConcreteSeries().toArray();
+  const scatterplotData = selectFirstSplitDatums(data);
+  const xExtent = getExtent(scatterplotData, xSeries);
+  const yExtent = getExtent(scatterplotData, ySeries);
+
+  const plottingStage = calculatePlottingStage(stage);
+  const yScale = d3.scale.linear().domain(yExtent).nice().range([plottingStage.height, 0]);
+  const xScale = d3.scale.linear().domain(xExtent).nice().range([0, plottingStage.width]);
+  const xTicks = pickTicks(xScale, 10);
+  const yTicks = pickTicks(yScale, 10);
+
+  return <div className="scatterplot-container" style={stage.getWidthHeight()}>
+    <span className="axisTitle" style={{ top: 10, left: 10 }}>{xSeries.title()}</span>
+    <span className="axisTitle" style={{ bottom: 145, right: 10 }}>{ySeries.title()}</span>
+    <svg viewBox={stage.getViewBox()}>
+      <GridLines orientation={"vertical"} stage={plottingStage} ticks={xTicks} scale={xScale} />
+      <GridLines orientation={"horizontal"} stage={plottingStage} ticks={yTicks} scale={yScale} />
+      <XAxis scale={xScale} stage={calculateXAxisStage(plottingStage)} ticks={xTicks} formatter={xSeries.formatter()}/>
+      <YAxis
+        stage={calculateYAxisStage(plottingStage)}
+        ticks={yTicks}
+        tickSize={TICK_SIZE}
+        scale={yScale}
+        formatter={ySeries.formatter()} />
```

Comment:
`formatter()` is deprecated

Leaving the comment to remember to refactor this.

Reply:
This is okay for axis, for data points use `formatValue`.